src/extras/auto-import.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Auto-import the first call to System.register 
+ */
+(function (global) {
+  var systemJSPrototype = global.System.constructor.prototype;
+  var systemInstantiate = systemJSPrototype.instantiate;
+  var systemRegister = systemJSPrototype.register;
+  var systemGetRegister = systemJSPrototype.getRegister;
+
+  // hook into System.register (but release it after first usage)
+  systemJSPrototype.register = function (deps, declare) {
+    // reset System.register for subsequent usage
+    systemJSPrototype.register = systemRegister;
+
+    // register normally
+    systemRegister.call(this, deps, declare);
+
+    // save the current url, and pass it to `importLastRegisterAs`
+    var scripts = document.getElementsByTagName("script");
+    var scriptUrl = scripts[scripts.length - 1].src || document.baseURI;

```

Comment:
Rather than falling back to `document.baseURI` we should just exit the function to skip the feature entirely in this case.

Reply:
We should also confirm that the script is not `defer`, `async` or `module`. And similarly exit the function to skip the feature in all of those cases.

Reply:
Indeed, `defer`, `async` and `module` should not trigger the auto-import. Can you elaborate why `document.baseURI` should as well? I envisioned this for the use case that a script would be inlined, eg.

```html
<script src="system.js"></script>
<script>
System.register([...], function(...) {
});
</script>
```
When auto-importing this script, we still need a reference to the current file, in order to resolve imports to other files. That's why I used `document.baseURI`, in the same way `import.meta` is known in an inline ES module as `document.baseURI`.

Reply:
Ahh I missed that this was the case you were handling... makes sense. Although note we have a better base URL handling here - https://github.com/systemjs/systemjs/blob/master/src/common.js#L13 although unfortunately because this is an extra it can't be shared....

Actually, you know what, let's make this a core feature and focus on getting the byte footprint down.

Reply:
Do you need me to move it out of the extra's?

If it's in the core, how would it get activated?